mesa/experimental/cell_space/discrete_space.py

```diff
@@ -73,3 +83,38 @@ def empties(self) -> CellCollection[T]:
     def select_random_empty_cell(self) -> T:
         """Select random empty cell."""
         return self.random.choice(list(self.empties))
+
+    # PropertyLayer methods
+    def add_property_layer(
+        self, property_layer: PropertyLayer, add_to_cells: bool = True
+    ):
+        """Add a property layer to the grid."""
+        if property_layer.name in self.property_layers:
+            raise ValueError(f"Property layer {property_layer.name} already exists.")
+        self.property_layers[property_layer.name] = property_layer
+        if add_to_cells:
+            for cell in self._cells.values():
+                cell.add_property_layer(property_layer)
+
+    def remove_property_layer(self, property_name: str, remove_from_cells: bool = True):
+        """Remove a property layer from the grid."""
+        del self.property_layers[property_name]
+        if remove_from_cells:
+            for cell in self._cells.values():
+                cell.remove_property_layer(property_name)
+
+    def set_property(
+        self, property_name: str, value, condition: Callable[[T], bool] | None = None
+    ):
+        """Set the value of a property for all cells in the grid."""
+        self.property_layers[property_name].set_cells(value, condition)
```

Comment:
please (for all) specify the arguments as well as passing a docstring.

in pycharm, you can specify which docstring standard you want to use to make this easy (https://www.jetbrains.com/help/pycharm/settings-tools-python-integrated-tools.html)

Reply:
First of all, very useful tool. And you know I love good tooling, so thanks for linking to it.

> (for all) specify the arguments

I don't know if I necessarily agree with this. Every function, method and class should have at least the oneliner and type-hinted input and output. But I also think that sometimes can be enough.

Comments/docstring are there to:
- Explain conceptual or higher level workings
- Explain unintuitive or complex code.

If there are no conceptual workings and the code is simple and intuitive, it's readable on its own. So I don't think it needs docstring in those cases.

I will reflect on these specific cases.

Reply:
Added quite some docstring